controlplane/pkg/remote/proxy_network_service_server/proxy_network_service_server.go

```diff
@@ -0,0 +1,149 @@
+package proxynetworkserviceserver
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/registry"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/nsmd"
+	"github.com/networkservicemesh/networkservicemesh/k8s/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"os"
+	"strings"
+
+	remote_connection "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/remote/connection"
+	remote_networkservice "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/remote/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/serviceregistry"
+)
+
+// Default values and environment variables of proxy connection
+const (
+	ProxyNsmdK8sAddressEnv         = "PROXY_NSMD_K8S_ADDRESS"
+	ProxyNsmdK8sAddressDefaults    = "pnsmgr-svc:5005"
+	ProxyNsmdK8sRemotePortEnv      = "PROXY_NSMD_K8S_REMOTE_PORT"
+	ProxyNsmdK8sRemotePortDefaults = "80"
+)
+
+type proxyNetworkServiceServer struct {
+	serviceRegistry serviceregistry.ServiceRegistry
+}
+
+// NewProxyNetworkServiceServer creates a new remote.NetworkServiceServer
+func NewProxyNetworkServiceServer(serviceRegistry serviceregistry.ServiceRegistry) remote_networkservice.NetworkServiceServer {
+	server := &proxyNetworkServiceServer{
+		serviceRegistry: serviceRegistry,
+	}
+	return server
+}
+
+func (srv *proxyNetworkServiceServer) Request(ctx context.Context, request *remote_networkservice.NetworkServiceRequest) (*remote_connection.Connection, error) {
+	logrus.Infof("ProxyNSMD: Received request from client to connect to NetworkService: %v", request)
+
+	destNsmName := request.Connection.DestinationNetworkServiceManagerName
+	dNsmName, dNsmAddress, err := utils.ParseNsmURL(destNsmName)
+	if err == nil {
```

Comment:
Why don't you handle `err != nil` here? It will reduce nesting for the code below